flutter/lib/src/sentry_widget.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+/// Key which is used to identify the [RepaintBoundary]
+final sentryWidgetGlobalKey = GlobalKey(debugLabel: 'sentry_widget');
+
+/// You can add screenshots of [child] to crash reports by adding this widget.
+/// Ideally you are adding it around your app widget like in the following
+/// example.
+/// ```dart
+/// runApp(SentryWidget(child: App()));
+/// ```
+///
+/// Remarks:
+/// - Depending on the place where it's used, you might have a transparent
+///   background.
+/// - Platform Views currently can't be captured.
+/// - It only works on Flutters Canvas Kit Web renderer. For more information
+///   see https://flutter.dev/docs/development/tools/web-renderers
+/// - You can only have one [SentryWidget] widget in your widget tree at all
+///   times.
+class SentryWidget extends StatefulWidget {
```

Comment:
> So the idea here is that SentryWidget is the root widget and we'd add more features on top of that, screenshots is just the first.

What I wanted to say is, that by splitting up individual integrations, which have to be added via the widget tree it's easy to configure what you want. And then the `SentryWidget` will be an umbrella widget for all widget tree integrations.

Right now, the screenshot widget is already the second widget tree integration (the asset bundle is the first), so it would make sense (at least for me) to name this `SentryScreenshot` or something.

> The difference is that if you use this name now, later its not StatefulWidget anymore but rather a BindingBase

I don't get what you mean by this. Regardless, even if a bindingbase will be used the widget is still needed. The thing that changes is that the users doesn't have to add it manually anymore.

Reply:
gotcha, that makes sense, lets call it `SentryScreenshotWidget` for now.